File: frontend/src/microservices/tour-group/objects/TourGroupMembershipTravelerResponse.ts
// 本文件定义 tour-group 模块的 `TourGroupMembershipTravelerResponse`，作为响应数据并提供 JSON 编解码。

export type TourGroupMembershipTravelerResponse = {
  membershipTravelerId: string
  membershipId: string
  travelerId: string
  status: string
  joinedAt: string
}
export const tourGroupMembershipTravelerResponseFromJson = (json: string): TourGroupMembershipTravelerResponse =>
  JSON.parse(json) as TourGroupMembershipTravelerResponse

export const tourGroupMembershipTravelerResponseToJson = (value: TourGroupMembershipTravelerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/tour-group/objects/TourGroupMembershipTravelerResponse.scala
// TourGroupPlannerModels 定义团体游模块的请求和响应模型。

package com.typesafe.travel.tourgroup.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TourGroupMembershipTravelerResponse(membershipTravelerId: String, membershipId: String, travelerId: String, status: String, joinedAt: String)
object TourGroupMembershipTravelerResponse:
  given sourceEncoder: Encoder[TourGroupMembershipTravelerResponse] = deriveEncoder
  given sourceDecoder: Decoder[TourGroupMembershipTravelerResponse] = deriveDecoder
